test/benchmark_http_client_test.cc

```diff
@@ -37,37 +38,81 @@ using namespace testing;
 
 namespace Nighthawk {
 
-class BenchmarkClientTestBase : public Envoy::BaseIntegrationTest,
-                                public TestWithParam<Envoy::Network::Address::IpVersion> {
+class PythonIntegrationTestBase : public TestWithParam<Envoy::Network::Address::IpVersion> {
 public:
-  BenchmarkClientTestBase()
-      : Envoy::BaseIntegrationTest(GetParam(), realTime(), BenchmarkClientTestBase::envoy_config),
-        api_(thread_factory_, store_, timeSystem(), file_system_),
-        dispatcher_(api_.allocateDispatcher()) {}
+  PythonIntegrationTestBase(const Envoy::Network::Address::IpVersion version)
+      : version_(version), pipe_(nullptr){};
+
+  void startPythonIntegrationWrapper(const bool use_https) {
+    std::string args;
 
-  static void SetUpTestCase() {
-    Envoy::Filesystem::InstanceImplPosix file_system;
-    envoy_config = file_system.fileReadToEnd(Envoy::TestEnvironment::runfilesPath(
-        "test/test_data/benchmark_http_client_test_envoy.yaml"));
-    envoy_config = Envoy::TestEnvironment::substitute(envoy_config);
+    if (GetParam() == Envoy::Network::Address::IpVersion::v4) {
+      args.append(" ipv4");
+    } else {
+      args.append(" ipv6");
+    }
+    args.append(" http");
+    if (use_https) {
+      args.append("s");
+    }
+    pipe_ = popen(Envoy::TestEnvironment::runfilesPath("integration/cpp_benchmark_client_server")
+                      .append(args)
+                      .c_str(),
+                  "r");
+    RELEASE_ASSERT(pipe_ != nullptr, "Failed to open pipe");
+    RELEASE_ASSERT(fgets(buffer_.data(), buffer_.max_size(), pipe_) != nullptr,
+                   "Expected more data");
+    RELEASE_ASSERT(absl::SimpleAtoi(buffer_.data(), &server_port_),
+                   "couldn't understand server_pid");
+    RELEASE_ASSERT(fgets(buffer_.data(), buffer_.max_size(), pipe_) != nullptr,
+                   "Expected more data");
+    RELEASE_ASSERT(absl::SimpleAtoi(buffer_.data(), &server_pid_),
+                   "couldn't understand server_pid");
+  }
+
+  void TearDown() override {
+    if (server_pid_ != 0) {
+      kill(server_pid_, SIGTERM);
+    }
+    if (pipe_ != nullptr) {
+      // We don't expect any output, lets print it when that happens
+      while (fgets(buffer_.data(), buffer_.max_size(), pipe_) != nullptr) {
```

Comment:
Generally bad form to use `fgets`, can you use C++ streams or something a bit safer?